File: LeetCode/1572. Matrix Diagonal Sum/elixir/solution.exs
defmodule Solution do
  @spec diagonal_sum(mat :: [[integer]]) :: integer
  def diagonal_sum(mat) do
    length = length(mat)

    mat
    |> Enum.with_index()
    |> Enum.reduce(0, fn {row, i}, acc ->
      if rem(length, 2) == 1 and div(length, 2) == i do
        acc + Enum.at(row, i)
      else
        acc + Enum.at(row, i) + Enum.at(row, length - 1 - i)
      end
    end)
  end
end

File: LeetCode/1572. Matrix Diagonal Sum/java/Solution.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int answer = 0;
        int length = mat.length;

        for (int i = 0; i < length; i++) {
            if (length % 2 == 1 && i == length / 2) {
                answer += mat[i][i];
            } else {
                answer += mat[i][i] + mat[i][length - 1 - i];
            }
        }
        return answer;
    }
}